### PR TITLE
Resolved multiple open issues

### LINKED ISSUE #9: Required inputfields
All input fileds for the CLI should probably be set to required.

### LINKED ISSUE #11: Replace io name
**What needs to be done:**
io class needs to change name avoid collsion

**Suggestions on how to get it done:**
Name it to something snassy. Run the tests. Integrate.

**What are the arguments for getting it done:**
Kills io.

**Task is considered finished when:**
All instances referring to the HCPinterface class has been replaced

### LINKED ISSUE #12: Add log2file and log entries
**What needs to be done:**

**Suggestions on how to get it done:**

**What are the arguments for getting it done:**

**Task is considered finished when:**

### LINKED ISSUE #16: Generalise ProgressBar
**What needs to be done:**

**Suggestions on how to get it done:**

**What are the arguments for getting it done:**

**Task is considered finished when:**

### LINKED ISSUE #17: Move transfer config settings out of config.ini
**What needs to be done:**
Remove config.ini and apply default settings to transfer config in HCPManager.

**Suggestions on how to get it done:**
Remove config.ini and move settings into HCPManager. Function calculating sha256 needs to know these values however. These could go into  file called "definitions.py" or alike.

**What are the arguments for getting it done:**
Issues with reading the config.ini when installing to conda env. Configparser might also present issues if parent package also uses configparser.
`configparser.NoSectionError: No section: 'hcp'` after fresh install.

**Task is considered finished when:**
config.ini is no longer relied upon and HCPManager.upload_file() and calculate_etag() function as normal.

### LINKED ISSUE #18: errror creating HCPmanager instance
Cannot establish a hcpmanager connection. Error msg below (removed/modified all credentials for security reasons):

PyDev console: starting.
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)] on win32
from HCPInterface.hcp import HCPManager
endpoint = "https://vgtn0xxxx43"
aws_access_key_id = "xxxx"
aws_secret_access_key = "xxxx"
hcpm = HCPManager(endpoint, aws_access_key_id, aws_secret_access_key)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "C:\Users\PELA09\PycharmProjects\GMS-uploader\venv\lib\site-packages\HCPInterface\hcp\hcp.py", line 134, in __init__
    self.transfer_config = TransferConfig(multipart_threshold=config.getint('hcp', 'size_threshold'),
  File "C:\Users\PELA09\AppData\Local\Programs\Python\Python38\lib\configparser.py", line 818, in getint
    return self._get_conv(section, option, int, raw=raw, vars=vars,
  File "C:\Users\PELA09\AppData\Local\Programs\Python\Python38\lib\configparser.py", line 808, in _get_conv
    return self._get(section, conv, option, raw=raw, vars=vars,
  File "C:\Users\PELA09\AppData\Local\Programs\Python\Python38\lib\configparser.py", line 803, in _get
    return conv(self.get(section, option, **kwargs))
  File "C:\Users\PELA09\AppData\Local\Programs\Python\Python38\lib\configparser.py", line 781, in get
    d = self._unify_values(section, vars)
  File "C:\Users\PELA09\AppData\Local\Programs\Python\Python38\lib\configparser.py", line 1149, in _unify_values
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'hcp'

### PR DESCRIPTION
**Pull request by sylvinite**
## Contents
Resolves #9 
Resolves #11 
Resolves #12 
Resolves #16 
Resolves #17
Resolves #18

### The What
Added credentials to gitignore
Fixed a progressbar induced test issue
Defaulted requirements to credentials file and bucket
CLI command now lets logs be written to file
Renamed io due to clash
Bumped test coverage to 60%
Generalised ProgressBar

Added cloned credentials.json in test repo
Moved freeze to requirements-dev. Removed HCPinterface mention from freeze.
Mock is part of the requirements-dev
Config.ini is now found regardless of how installed
Added the worlds simplest connection test

### This [update](https://semver.org/) is:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### Installation and initiation
* `git clone git@github.com:genomic-medicine-sweden/HCPInterface.git`
* `cd HCPInterface && git checkout <BRANCH>`
* `bash setup.sh`
* `source activate hpcenv`

### Tests
* `pytest tests/`

### Expected outcome:
* Pytest resolves without crashes

## Confirmations:
- [ ] Code reviewed by @octocat
- [x] Code tested by @sylvinite

### CODE REVIEW

**Comment by pbiology** [comment]
Tried running all tests in the NGP
Did get the following error during setup
`Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.25.11-py2.py3-none-any.whl (127 kB)
     |████████████████████████████████| 127 kB 91.5 MB/s
Collecting six>=1.5
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: six, urllib3, python-dateutil, jmespath, docutils, botocore, s3transfer, idna, chardet, requests, click, boto3
Successfully installed boto3-1.12.34 botocore-1.15.49 chardet-3.0.4 click-7.1.2 docutils-0.15.2 idna-2.10 jmespath-0.10.0 python-dateutil-2.8.2 requests-2.25.0 s3transfer-0.3.7 six-1.16.0 urllib3-1.25.11
Processing /home/xsylis/code/gms/HCPInterface
ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/home/xsylis/code/gms/HCPInterface'`

**Comment by pbiology** [comment]
When running the pytest I got 2 errors.
1. The module "mock" not existing, installing it in the conda env fixed that.
2. The file tests/credentials.json was not included.

File: HCPInterface/preproc/preproc.py [hunk 1]
```diff
@@ -57,3 +57,19 @@ def generate_tagmap(fn, tag, out="{}/meta-{}.json".format(os.getcwd(), TIMESTAMP
     md.write(json.dumps(mdict, indent=4))
     md.close()
     log.debug(f'Generated metadata file {out}')
+
```
Comment (pbiology) [inline HCPInterface/preproc/preproc.py:61]: This file in general could do with some more in-line comments

File: tests/test_cmds.py [hunk 2]
```diff
@@ -53,10 +53,10 @@ def test_upload(runner):
     res = runner.invoke(root, cmd.split())
     assert res.exit_code == 0
 
-#def test_search(runner):
-#    cmd = "-b ngs-test -c {0} search -q {1}".format(credentials_path, f1target)
-#    res = runner.invoke(root, cmd.split())
-#    assert res.exit_code == 0 
+def test_search(runner):
+    cmd = "-b ngs-test -c {0} search -q {1}".format(credentials_path, f1target)
```
Comment (pbiology) [inline tests/test_cmds.py:61]: Might be a future issue if ngs-test gets moved or emptied etc

File: tests/test_hcp.py [hunk 3]
```diff
@@ -22,14 +22,14 @@
 class TestProcess(unittest.TestCase):
 
     def test01_upload_file(self):
-        self.assertIsNone(hcpm.upload_file(f"{testWD}/data/test_reads_R1.fasterq", "unittest/test_reads_R1.fasterq"))
+        self.assertIsNone(hcpm.upload_file(f"{testWD}/data/test_reads_R1.fasterq", "unittest/test_reads_R1.fasterq",silent=True))
```
Comment (pbiology) [inline tests/test_hcp.py]: Will these petagene compressed test files work on a system that doesn't have petagene installed?

File: HCPInterface/preproc/preproc.py (hunk 1 above)
Comment (sylvinite) [reply HCPInterface/preproc/preproc.py:61]: Every function has desc. Python code is pretty descriptive. Not sure what is requested

File: tests/test_cmds.py (hunk 2 above)
Comment (sylvinite) [reply tests/test_cmds.py:61]: Clarified in source

File: tests/test_hcp.py (hunk 3 above)
Comment (sylvinite) [reply tests/test_hcp.py]: It transfers something. But idk. Can just replace with fastq if desired

**Comment by sylvinite** [comment]
Added cloned credentials.json in test repo
Moved freeze to requirements-dev. Removed HCPinterface mention from freeze.
Mock is part of the requirements-dev

File: setup.py [hunk 4]
```diff
@@ -16,4 +16,8 @@
     entry_points={
         "console_scripts": ["hcpi=HCPInterface.cli.base:root"],
     },
+    data_files=[('config', ['config.ini']),
+                ('reference', ['reference/covidMetadataAllowedValues.csv', 'reference/covidMetadataTemplate.csv']),
+                ('testdata', ['tests/data/test_reads_R1.fasterq','tests/data/test_reads_R1.fastq.gz','tests/data/test.json'])
```
Comment (pbiology) [inline setup.py:21]: Just a thought. Is this dataset completely cleaned from any human DNA? So we don't end up in a situation where we are using any identifiable data in our tests.

File: tests/simpleConnectionTest.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from HCPInterface import WD
+from HCPInterface.hcp import HCPManager
```
Comment (pbiology) [inline tests/simpleConnectionTest.py:2]: So is it HCPinterface or NGPinterface? Correct name going forward?

File: setup.py (hunk 4 above)
Comment (sylvinite) [reply setup.py:21]: Yea its just random garbo that looks like fastq

File: tests/simpleConnectionTest.py (hunk 5 above)
Comment (sylvinite) [reply tests/simpleConnectionTest.py:2]: Will do it in a sep PR since itll be a bit of a pain

**Pull request merged**